public/pages/CreateMonitor/components/ClusterMetricsMonitor/ClusterMetricsMonitor.js

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useState } from 'react';
+import _ from 'lodash';
+import {
+  EuiButton,
+  EuiButtonEmpty,
+  EuiCodeEditor,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiLink,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+import { isInvalidApiPath } from '../../../../utils/validate';
+import { FormikComboBox, FormikFieldText } from '../../../../components/FormControls';
+import {
+  API_PATH_REQUIRED_PLACEHOLDER_TEXT,
+  EMPTY_PATH_PARAMS_TEXT,
+  REST_API_REFERENCE,
+  API_TYPES,
+} from './utils/clusterMetricsMonitorConstants';
+import {
+  getApiPath,
+  getExamplePathParams,
+  isInvalidApiPathParameter,
+  validateApiPathParameter,
+} from './utils/clusterMetricsMonitorHelpers';
+import { FORMIK_INITIAL_VALUES } from '../../containers/CreateMonitor/utils/constants';
+
+const setApiType = (form, apiType) => {
+  const pathParams = _.get(form, 'uri.path_params', FORMIK_INITIAL_VALUES.uri.path_params);
+  form.setFieldValue('uri.api_type', apiType);
+  form.setFieldValue('uri.path', getApiPath(_.isEmpty(pathParams), apiType));
+};
+
+const renderModal = (closeModal, prevApiType, selectedApiType, form) => {
```

Comment:
For this clear trigger modal, I see that this is used when the cluster metrics monitor is changing the api type, right? Do you think this modal can be reused for other monitor update conditions such as: When user has trigger defined and changing from query monitor to bucket monitor, show this clear trigger condition modal? Or do we already have the logic implemented for this case.

Reply:
We **do not** currently have similar logic in place when changing from query-level to bucket-level. When implementing the bucket-level UX, the were a few short discussions about whether and when it would be appropriate to clear the form, but we left that as an open topic. This is a good point though; we should revisit this. I made this issue to track this discussion https://github.com/opensearch-project/alerting-dashboards-plugin/issues/188